### PR TITLE
Added progress bar for online search and lychee filter

### LINKED ISSUE #21: Show progress when searching online

### PR DESCRIPTION
**Pull request by TheRustyPickle**
**What changed:**
* Add new dep `indicatif` for creating progress bars
* Added progress bars on online search and on lychee filter

Resolves #21. The progress bar style and the color can be changed. Let me know if you have something else in mind.

### CODE REVIEW

File: src/web_crawler/twir_crawler.rs [hunk 1]
```diff
@@ -272,3 +294,15 @@ impl TwirCrawler {
         Ok(())
     }
 }
+
+fn get_progress_bar(text: &str) -> ProgressBar {
+    let progress_bar = ProgressBar::new(100);
```
Comment (vlad-onis) [inline src/web_crawler/twir_crawler.rs:299]: Sorry to be pedantic here, maybe it's no value - let me know what you think. We can add a "utils" module if you think it's a good idea. Put this function there and so we can define constants there to replace magic numbers. Something along the lines of const 

```rust
const PROGRESS_BAR_LENGTH: u64 = 100;
```

File: src/web_crawler/twir_crawler.rs [hunk 2]
```diff
@@ -272,3 +294,15 @@ impl TwirCrawler {
...
+    let style = ProgressStyle::default_bar()
+        .template("{prefix} [{bar:40.cyan/blue}] {percent}% {eta} {msg}")
+        .unwrap()
```
Comment (vlad-onis) [inline src/web_crawler/twir_crawler.rs:302]: Can this one fail?

File: src/web_crawler/twir_crawler.rs [hunk 3]
```diff
@@ -148,11 +158,23 @@ impl TwirCrawler {
...
+        let progress_bar = get_progress_bar("Checking Issues");
+
         let limit = limit as usize;
 
         for (index, issue) in issues_and_titles.into_iter().enumerate() {
             found_resources.append(&mut self.parse_page(&issue.link, sentence).await?);
 
+            current_bar_value += bar_value;
+
+            let int_bar_value = current_bar_value.round() as u64;
```
Comment (vlad-onis) [inline src/web_crawler/twir_crawler.rs:175]: the name says int_bar_value but it's a u64. I suggest a rename here, could be something like "rounded_bar_value", wdyt?

**Comment by vlad-onis** [comment]
Thanks again for the contribution! I really like it 🚀🦀
I've left a few comments but they are all neat picks. Let me know what you think about them but it is otherwise a nice MR and ready to merge.

File: src/web_crawler/twir_crawler.rs (hunk 2 above)
Comment (TheRustyPickle) [reply src/web_crawler/twir_crawler.rs:302]: It can fail on invalid placeholders (values inside {})

File: src/web_crawler/twir_crawler.rs (hunk 1 above)
Comment (TheRustyPickle) [reply src/web_crawler/twir_crawler.rs:299]: Good point. A utils mod would be suitable here. Will create a utils mod and move it there.

File: src/web_crawler/twir_crawler.rs (hunk 2 above)
Comment (vlad-onis) [reply src/web_crawler/twir_crawler.rs:302]: All good then, no need to wrap it in a Result, it will probably be uglier like that.

**Comment by TheRustyPickle** [comment]
Let me know if I missed anything. Thanks!

**Pull request merged**